### ISSUE TITLE
[Bug]: lazyCompilation does not trigger invalidation for initial page

### ISSUE THREAD

**Issue by kwonoj** [web-infra-dev/rspack#8305]
### System Info

npx envinfo --system --npmPackages '@rspack/*' --binaries --browsers

  System:
    OS: macOS 14.7
    CPU: (12) arm64 Apple M3 Pro
    Memory: 203.03 MB / 36.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.15.1 - ~/.local/state/fnm_multishells/1121_1730396689623/bin/node
    Yarn: 4.5.0 - ~/.local/state/fnm_multishells/1121_1730396689623/bin/yarn
    npm: 10.7.0 - ~/.local/state/fnm_multishells/1121_1730396689623/bin/npm
  Browsers:
    Chrome: 130.0.6723.70
    Safari: 18.0.1
  npmPackages:
    @rspack/cli: ^1.1.0-beta.0 => 1.1.0-beta.0
    @rspack/core: ^1.1.0-beta.0 => 1.1.0-beta.0
    @rspack/plugin-react-refresh: 1.0.0-beta.5 => 1.0.0-beta.5

### Details

Setting lazycompilation option

```
// doesn't matter either entries true | false or imports true | false, as long as one or both of them are enabled
{
          entries: true, 
          imports: false,
          backend: {
            server: {
              key: fs.readFileSync(PROXY_CERT.keyPath, 'utf-8'),
              cert: fs.readFileSync(PROXY_CERT.certPath, 'utf-8'),
            },
          } as any,
        }
```

makes compiler watch invalidation does not trigger the compilation. It shows initial path is found and will be compiled

```
 [LazyCompilationBackend] /Users/ohjoong.kwon/github/redacted/node_modules/whatwg-fetch/fetch.js is now in use and will be compiled.
    [LazyCompilationBackend] builtin:react-refresh-loader!builtin:swc-loader%3F%3FruleSet%5B1%5D.rules%5B1%5D.use%5B0%5D!/Users/ohjoong.kwon/github/redacted/src/command/index.js is now in use and will be compiled.
```

But it stops there. Peeking through, it looks like `invalidate` is not triggered by hitting here (https://github.com/web-infra-dev/rspack/blob/0db8b9441a8bf9447ce11cc69292df773482cec8/packages/rspack/src/Watching.ts#L206) which seems to be set in here (https://github.com/web-infra-dev/rspack/blob/0db8b9441a8bf9447ce11cc69292df773482cec8/packages/rspack/src/MultiCompiler.ts#L498)

but not sure why watching is blocked. Is there additional log to configure to see what's going on? I set below for the logging:

```
infrastructureLogging: {
    colors: true,
    level: 'verbose',
    debug: true,
  },
```

### Reproduce link

private code :/

### Reproduce Steps

Setup lazycompilation and run rspack, open the initial page

**Comment by kwonoj** [comment]
It looks like if I manually touch one of the entry file (`Users/ohjoong.kwon/github/redacted/src/command/index.js`) looks like invalidation start to kicks in.

**Comment by kwonoj** [comment]
@JSerFeng fyi, I think I was able to narrow down repro

https://github.com/kwonoj/rspack-lazy-try-import/blob/main/src/index.js#L4-L12

If a promise returned by dynamic import is awaited later, seems it is stuck in the watcher and lazycompilation doesn't trigger the invalidation.

**Comment by kwonoj** [comment]
Actually repro seems wrong :/ it was reproducible on the actual codebase, but in the repro module's being loaded.

**Comment by JSerFeng** [comment]
> If a promise returned by dynamic import is awaited later, seems it is stuck in the watcher and lazycompilation doesn't trigger the invalidation.

Yeah, I tried a demo with endless promise, seems fine.

**Comment by kwonoj** [comment]
Any diagnostic I can run to grab more details? It's not easy to create a repro since I don't have a clue where / why it happens even.

**Comment by JSerFeng** [comment]
Are you using multiCompiler ? seems it's the only place to modify this, sorry we don't have enough trace here to debug

**Comment by JSerFeng** [comment]
Cause if compiler A is running, and compiler B depends on A, compiler B is blocked

I don't know if this is the reason, I'll give that a try when I'm available

**Comment by kwonoj** [comment]
Yes, we have multi compiler config. A depends on B, but B does not depends on anything. (and seeing A is waiting indefinitely)

**Comment by JSerFeng** [comment]
Got it, I'm trying to reproduce this
